analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```

Comment:
could you also the examples inside the RSPEC? To have the in the UTs

Reply:
As far as I can tell they are already there.

The first case:
https://github.com/SonarSource/sonar-dotnet/pull/4874/files#diff-120f54707b417f7640b7722b69ca748f7d88d8a53b8fe03450b82d22a5990870R21

The second case: 
https://github.com/SonarSource/sonar-dotnet/pull/4874/files#diff-120f54707b417f7640b7722b69ca748f7d88d8a53b8fe03450b82d22a5990870R162

I've added one more with a null check to be extra safe 😄

Reply:
what I see is the Noncompliant cases from the RSPEC.

I meant the _compliant_ cases where LINQ expressions are used.

Reply:
what I see is the Noncompliant cases from the RSPEC.

I meant the _compliant_ cases where LINQ expressions are used - otherwise how can the reader see that the "compliant" cases from the RSPEC are truly compliant?

Reply:
Yes, my bad. I've misread your comment. I've added the cases and I will add a new PR on RSPEC to update them a bit since the current version does not compile :)

Reply:
See here: https://github.com/SonarSource/rspec/pull/337